shepherd_bms/src/stateMachine.cpp

```diff
@@ -319,12 +320,31 @@ bool StateMachine::chargingCheck(AccumulatorData_t *bmsdata)
 
 bool StateMachine::balancingCheck(AccumulatorData_t *bmsdata)
 {
+	static Timer balanceTimer;
+
 	if (!compute.isCharging()) return false;
 	if (bmsdata->max_temp.val > MAX_CELL_TEMP_BAL) return false;
 	if (bmsdata->max_voltage.val <= (BAL_MIN_V * 10000)) return false;
 	if(bmsdata->delt_voltage <= (MAX_DELTA_V * 10000)) return false;
+	//if these all return true, then we should balance
 
-	return true;
+	if (balanceTimer.isTimerExpired() && isBalancing == false) //if timer is done/off and not balancing, balance and start timer
+	{
+		isBalancing = true;
+		balanceTimer.startTimer(BALANCE_TIME);	//start balancing for 5 min
+	}
+	if (balanceTimer.isTimerExpired() && isBalancing == true)	//if timer is done/off and we're balancing, pause balancing & start 1 min timer
+	{
+		isBalancing = false;
+		balanceTimer.startTimer(BALANCE_TIMEOUT);	//pause balancing for 1 min
```

Comment:
I think that this will just keep upping the timer and it'll never go off, right?

Reply:
Do I need to call the reset timer function?

Reply:
@nwdepatie i dont agree